### PR TITLE
Feature/aws prompts

### PR DESCRIPTION
**Pull request by ggotti**
Added loading status functionality to AWS operations
Connected prompts to AWS to retrieve valid values
Moved AWS operations which rely on filesystem to the end() section. I decided to do this instead of using the Virtual File system, as we need to recompile the application if we're going to deploy (so we can use the Spring Cloud Stuff we're injecting earlier)

### CODE REVIEW

File: generators/more-aws/README.md [hunk 1]
```diff
@@ -16,5 +16,5 @@ The token received by ECR to authenticate is structured as follow. We'll only ta
 ### Dependencies
 
 ```
-npm i --save-dev aws-sdk@^2.167.0 progress@2.0.0
+npm i --save-dev aws-sdk@^2.167.0 progress@2.0.0 cli-spinners@1.1.0
```
Comment (carvallegro) [inline generators/more-aws/README.md]: isn't it `ora@1.3.0` instead?

File: generators/more-aws/index.js [hunk 2]
```diff
@@ -54,6 +73,54 @@ module.exports = class extends BaseGenerator {
...
+                    .then(() => done())
+                    .catch(() => done('Error while loading the AWS library'));
+            },
+            checkAwsCredentials() {
+                if (this.abort) return;
+                const done = this.async();
+                let profile = process.env.AWS_PROFILE;
+                if (!profile) {
+                    profile = 'default';
+                    // this.log(chalk.white(`No profile found in AWS_PROFILE env var, using default : ${chalk.bold(profile)}`));
```
Comment (carvallegro) [inline generators/more-aws/index.js]: Delete or uncomment

File: generators/more-aws/index.js [hunk 3]
```diff
@@ -54,6 +73,54 @@ module.exports = class extends BaseGenerator {
...
+                    });
+            },
+            initAwsStuff() {
+                awsClient.initAwsStuff();
+            },
+            fetchRegion() {
+                // get the region using awsClient.getRegion
+                if (this.abort) return;
+                const done = this.async();
+                this.facts = {
```
Comment (carvallegro) [inline generators/more-aws/index.js]: Just stick AWS somewhere in this variable.

File: generators/more-aws/index.js [hunk 4]
```diff
@@ -160,17 +210,48 @@ module.exports = class extends BaseGenerator {
...
-                    vpc: 'vpc-ee3a5796',
-                    vpc_cidr: '172.31.0.0/16',
-                    vpc_subnets: ['subnet-2779a208', 'subnet-e891f6a3'],
                     taskCount: 1,
-                    databaseSize: 'db.t2.small',
                     databaseUsername: 'monolithuser',
                     databasePassword: 'fsdkdfljkdsfjdfsjpcvxcv4fgdfg4dfbvbtg'
                 };
             },
+            addAWSDependencies() {
```
Comment (carvallegro) [inline generators/more-aws/index.js]: addAWSSpringDependencies ?

File: generators/more-aws/prompts.js [hunk 5]
```diff
@@ -48,6 +54,35 @@ function askTypeOfApplication() {
...
+ * Ask user what type of Region is to be created?
+ */
+function askRegion() {
+    if (this.abort) return null;
+    const done = this.async();
+    const prompts = [
+        {
+            type: 'list',
+            name: 'region',
+            message: 'What region?',
```
Comment (carvallegro) [inline generators/more-aws/prompts.js]: Which region?

File: generators/more-aws/prompts.js [hunk 6]
```diff
@@ -58,8 +93,8 @@ function askCloudFormation() {
         {
             type: 'input',
             name: 'cloudFormationName',
-            message: 'Please enter your CloudFormation Stack name.',
-            default: this.aws.s3BucketName,
+            message: 'Please enter your stack\'s name. (must be unique within a region)',
+            default: (this.aws.cloudFormationName) ? this.aws.cloudFormationName : this.baseName,
```
Comment (carvallegro) [inline generators/more-aws/prompts.js]: I'm pretty sure you can do `this.aws.cloudFormationName || this.baseName`
It's a known JS feature

File: generators/more-aws/README.md (hunk 1 above)
Comment (ggotti) [reply generators/more-aws/README.md]: 🌮

File: generators/more-aws/index.js (hunk 2 above)
Comment (ggotti) [reply generators/more-aws/index.js]: 🌮

File: generators/more-aws/index.js (hunk 3 above)
Comment (ggotti) [reply generators/more-aws/index.js]: 🌮

File: generators/more-aws/index.js (hunk 4 above)
Comment (ggotti) [reply generators/more-aws/index.js]: 🌮

File: generators/more-aws/prompts.js (hunk 5 above)
Comment (ggotti) [reply generators/more-aws/prompts.js]: 🌮

File: generators/more-aws/prompts.js (hunk 6 above)
Comment (ggotti) [reply generators/more-aws/prompts.js]: 🌮

**Pull request merged**